### PR TITLE
HA VPN interop template

### PR DESCRIPTION
**Pull request by ashishverm**

### CODE REVIEW

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+    + There are notes in the template for you, the author, that explain where you need to
+      make changes. These notes are enclosed in brackets (&lt; &gt;).
+    + The template contains placeholders for things like the vendor and 
+      product name. These are also enclosed in brackets—for example, 
+      every place you see `<vendor-name>` and `<product-name>`, 
+      substitute approriate names.
+    + After you've made appropriate updates, _remove_ bracketed content.
+    + Remove these instructions.
+    + Because this is a template for a variety of setups, it might contain
+      content that isn't releavent to your scenario. Remove (or update)
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Typo: relevant

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+      any sections that don't apply to you.
+1. Fork the [GoogleCloudPlatform/community/](https://github.com/GoogleCloudPlatform/community/) repo.
+4. In your fork, add a new folder named `/tutorials/[YOUR_TUTORIAL]`. For the
+   folder name, use hyphens to separate words. We recommend that you 
+   include a product name in the folder name, such as `https-load-balancing-ingix`.
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Possibly you should add the directory name of a VPN tutorial instead of `https-load-balancing-ingix.`

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 3]
```diff
@@ -0,0 +1,836 @@
...
+        - [Configure the IPsec static virtual tunnel interface (SVTI)](#configure-the-ipsec-static-virtual-tunnel-interface-svti)
+    - [Configuring the dynamic routing protocol](#configuring-the-dynamic-routing-protocol)
+    - [Saving the configuration](#saving-the-configuration)
+    - [Testing the configuration](#testing-the-configuration)
+- [Troubleshooting IPsec on \<vendor-name>\<product-name>](#troubleshooting-ipsec-on-vendor-nameproduct-name)
+- [Reference documentation](#reference-documentation)
+    - [GCP documentation](#gcp-documentation)
+    - [\<vendor-name>\<product-name> documentation](#vendor-nameproduct-name-documentation)
+
+<Put trademark statements here>: <vendor terminology> and the <vendor> logo are
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: In the doc view, the text is truncated to the left of the word "Logo."

**Comment by ashishverm** [comment]
I signed it!

File: resources/templates/ha-vpn-interop-guide-template/index.md (hunk 1 above)
Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Done

File: resources/templates/ha-vpn-interop-guide-template/index.md (hunk 2 above)
Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Done

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 4]
```diff
@@ -0,0 +1,836 @@
...
+<This is some sample terminology. Add any terminology to this section that needs
+explanation.>
+
+-  **GCP VPC network**—A single virtual network within a single GCP project.
+-  **On-premises gateway**—The VPN device on the non-GCP side of the
+connection, which is usually a device in a physical data center or in
+another cloud provider's network. GCP instructions are written from the
+point of view of the GCP VPC network, so "on-premises gateway" refers to the
+gateway that's connecting _to_ GCP.
+-  **External IP address** or **GCP peer address**—a single static IP address
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: With HA VPN , one external address for each HA VPN interface are automatically created when you create the gateway. You don't create External IPs manually as you do for Classic VPN.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 5]
```diff
@@ -0,0 +1,836 @@
...
+within a GCP project that exists at the edge of the GCP network.
+-  **Dynamic routing**—GCP dynamic routing for VPN using the
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: HA VPN supports only dynamic routing.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 6]
```diff
@@ -0,0 +1,836 @@
...
+
+The \<vendor-name>\<product-name> equipment used in this guide is as follows:
+
+-  Vendor — <vendor-name>
+-  Model — <model name>
+-  Software release — <full software release name>
+
+## Before you begin
+
+1. Review information about how [dynamic routing](https://cloud.google.com/vpn/docs/concepts/choosing-networks-routing#dynamic-routing) works in GCP.
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Our convention is to spell out the first instance of GCP as "Google Cloud Platform."

File: resources/templates/ha-vpn-interop-guide-template/index.md (hunk 4 above)
Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: I am thinking to remove this section. We are explaining all the terms in the table below (Configuration parameters and values section) already. Thoughts ?

**Comment by jschneider11** [comment]
Ashish, I'll finish up reviewing this PR on Monday.

**Comment by jschneider11** [comment]
I'd keep it, because I think the terminology section goes into more detail
about the function of each resource.
If you want to move a short description into the table, that would work.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 7]
```diff
@@ -0,0 +1,836 @@
...
+
+<tr>
+<td>External static IP address for the second internet interface of <vendor
+name><product-name></td>
+<td><code>[ON_PREM_GW_IP_1]</code> </td>
+<td>For example, <code>199.203.248.182</code></td>
+</tr>
+
+<tr>
+<td>HA VPN Gateway</td>
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: I might split this into a second table and label it VPN gateway information.
You're also missing the autogenerated external IPs for each interface of the HA VPN gateway.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 8]
```diff
@@ -0,0 +1,836 @@
...
+
+1.Add a new BGP interface to the Cloud Router.
+
+    gcloud compute routers add-interface [ROUTER_NAME] \
+    --interface-name [ROUTER_INTERFACE_NAME_0] \
+    --mask-length [MASK_LENGTH] \
+    --vpn-tunnel [TUNNEL_NAME_0] \
+    --region [REGION]
+    
+The command output should look similar to the following example:
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Please replace lines 541 to 547 with command output from the original document.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 9]
```diff
@@ -0,0 +1,836 @@
...
+
+    gcloud compute routers add-bgp-peer [ROUTER_NAME] \
+    --peer-name [PEER_NAME] \
+    --peer-asn [PEER_ASN] \
+    --interface [ROUTER_INTERFACE_NAME_0] \
+    --region [REGION] \
+    
+ The command output should look similar to the following example:   
+ 
+    gcloud compute routers add-bgp-peer router-a \
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Please replace this with command output from the original document.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 10]
```diff
@@ -0,0 +1,836 @@
...
+
+    gcloud compute routers add-interface [ROUTER_NAME] \
+    --interface-name [ROUTER_INTERFACE_NAME_1] \
+    --mask-length [MASK_LENGTH] \
+    --vpn-tunnel [TUNNEL_NAME_1] \
+    --region [REGION]
+
+The command output should look similar to the following example:
+
+    gcloud compute routers add-interface router-a \
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Please replace this with command output from the original document.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 11]
```diff
@@ -0,0 +1,836 @@
...
+    
+1.Add a BGP peer to the interface for the second tunnel.
+
+    gcloud compute routers add-bgp-peer [ROUTER_NAME] \
+    --peer-name [PEER_NAME] \
+    --peer-asn [PEER_ASN] \
+    --interface [ROUTER_INTERFACE_NAME_1] \
+    --region [REGION] \
+
+The command output should look similar to the following example:
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Please replace this with command output from the original document.

File: resources/templates/ha-vpn-interop-guide-template/index.md (hunk 4 above)
Comment (jschneider11) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: The explanations aren't entirely the same. If you can fit them into the table, that might work.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 12]
```diff
@@ -0,0 +1,836 @@
...
+    --peer-external-gateway peer-gw \
+    --peer-external-gateway-interface 1  \
+    --region us-central1 \
+    --ike-version 2 \
+    --shared-secret mysharedsecret \
+    --router router-a \
+    --vpn-gateway ha-vpn-gw-a \
+    --interface 1
+    
+### Create Cloud Router interfaces and BGP peers
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: The step numbering in this section is showing up as "1." for every step.  I suspect that's due to Markdown indentation or spacing issues.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 13]
```diff
@@ -0,0 +1,836 @@
...
+       result.bgpPeerStatus[].ipAddress, result.bgpPeerStatus[].peerIpAddress)'
+    
+    result.bgpPeerStatus[0].ipAddress:     169.254.174.249
+    result.bgpPeerStatus[0].name:          peer-a
+    result.bgpPeerStatus[0].peerIpAddress: 169.254.174.250
+    result.bgpPeerStatus[1].ipAddress:     169.254.36.85
+    result.bgpPeerStatus[1].name:          peer-b
+    result.bgpPeerStatus[1].peerIpAddress: 169.254.36.86
+
+    gcloud compute routers describe [ROUTER_NAME] \
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: This is a command for listing the detailed output of the single Cloud Router you created for this tutorial. If you want to show that output, first add some intro text about this and add the command output from the original document after the command.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 14]
```diff
@@ -0,0 +1,836 @@
...
+    --region [REGION]
+
+### Configure firewall rules
+
+Configure firewall rules to allow inbound traffic from the on-premises
+network subnets. You must also configure the on-premises network firewall to
+allow inbound traffic from your VPC subnet prefixes.
+
+    gcloud compute firewall-rules create [VPN_RULE] \
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: This should actually be &lt;var&gt;VPN_RULE_NAME&lt;/var&gt;.  If you want to use the slashes instead as shown below, that would also work:

 \<vendor-name>\<product-name>

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 15]
```diff
@@ -0,0 +1,836 @@
...
+    --network [NETWORK] \
+    --allow tcp,udp,icmp \
+    --source-ranges [IP_ON_PREM_SUBNET]
+    
+The command output should look similar to the following example:    
+    
+     gcloud compute firewall-rules create network-a-to-on-prem \
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Please replace this with command output from the original document.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 16]
```diff
@@ -0,0 +1,836 @@
...
+    --network network-a \
+    --allow tcp,udp,icmp \
+    --source-ranges 192.168.1.0/24
+    
+## Configure the \<vendor-name>\<vendor product> side
+
+<This section includes sample tasks that describe how to configure the
+on-premises side of the VPN gateway configuration using <vendor-name>
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: &lt;vendor-name&gt; is not appearing in the rendered document. I suspect you need to use &lt; and &gt; to make the angle brackets show up.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 17]
```diff
@@ -0,0 +1,836 @@
...
+equipment.>
+
+### Creating the base network configuration
+
+<The sample wording below assumes that you are showing the configuration steps
+in the configuration code snippet below. If not, list the steps needed.>
+
+Follow the procedure listed in the configuration code snippet below to create
+the base Layer 3 network configuration of <vendor-name>. Note the following:
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Same issue with vendor-name not showing up as in line 638.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 18]
```diff
@@ -0,0 +1,836 @@
...
+Follow the procedures in this section to create the base VPN configuration.
+
+<This section contains outlines of subsections for different aspects of
+configuring IPsec and IKE on the vendor side. Fill in the sections that are
+relevant to the current configuration, and remove any sections that don't
+apply.>
+
+### GCP-compatible settings for IPSec and IKE
+
+ [GCP support ciphers](https://cloud.google.com/vpn/docs/how-to/configuring-peer-gateway#configuring_ike).
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Ciphers supported by GCP.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 19]
```diff
@@ -0,0 +1,836 @@
...
+
+<Insert the instructions for creating the IKE proposal and policy here. Below
+are some examples of IKE algorithms to specify as part of the instructions.>
+
+-  **Encryption algorithm**—< list required algorithms here>
+-  **Integrity algorithm**—< list required algorithms here>
+-  **Diffie-Hellman group—**< list required group here>
+
+```
+<insert configuration code snippet here>
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Capitalize "insert" here an in subsequent instructions.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 20]
```diff
@@ -0,0 +1,836 @@
...
+```
+
+#### Configure the IPsec security association (SA)
+
+< insert the instructions for creating the IPsec SA here. Below is an example of
+parameters to set.>
+
+-  **IPsec SA replay window-size**—Set this to 1024, which is the
+recommended value for <vendor-name><product-name>.
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: vendor-name  product-name not showing up, as mentioned previously.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 21]
```diff
@@ -0,0 +1,836 @@
...
+
+### Configuring the dynamic routing protocol
+
+Follow the procedure in this section to configure dynamic routing for traffic
+through the VPN tunnel or tunnels using the BGP routing protocol.
+
+< insert the instructions for configuring dynamic routing here. Below are some
+examples of parameters to set.>
+
+BGP timers are adjusted to provide more rapid detection of outages.
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Is this something the customer should do? If so, I'd reword this to < insert instructions for setting BGP timers x, y, and z settings to x, y, and z values>. In other words, you need to be specific about what timers and what values to set on the peer side, unless this is determined entirely by the customer.  Plus add inserting a config code snippet for this, as well.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 22]
```diff
@@ -0,0 +1,836 @@
...
+
+It's important to test the VPN connection from both sides of a VPN tunnel. For either side, make sure that the subnet that a machine or virtual machine is located in is being forwarded through the VPN tunnel.
+
+First, create VMs on both sides of the tunnel. Make sure that you configure the
+VMs on a subnet that will pass traffic through the VPN tunnel.
+
+-  Instructions for creating virtual machines in Compute Engine are located
+in the 
+[Getting Started Guide](https://cloud.google.com/compute/docs/quickstart).
+-  Instructions for creating virtual machine for <vendor-name><product-name>
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Same issue with the angle brackets on these two lines. Change them to &lt; and to &gt; to get the text inside them to show up on the rendered page.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 23]
```diff
@@ -0,0 +1,836 @@
...
+
+```
+<insert configuration code snippet here>
+```
+
+### Testing the configuration
+
+It's important to test the VPN connection from both sides of a VPN tunnel. For either side, make sure that the subnet that a machine or virtual machine is located in is being forwarded through the VPN tunnel.
+
+First, create VMs on both sides of the tunnel. Make sure that you configure the
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: I would number these steps.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 24]
```diff
@@ -0,0 +1,836 @@
...
+VMs on a subnet that will pass traffic through the VPN tunnel.
+
+-  Instructions for creating virtual machines in Compute Engine are located
+in the 
+[Getting Started Guide](https://cloud.google.com/compute/docs/quickstart).
+-  Instructions for creating virtual machine for <vendor-name><product-name>
+platforms are located at <link here>.
+
+After VMs have been deployed on both the GCP and <vendor-name><product-name>
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: After you have deployed VMs on both the...

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 25]
```diff
@@ -0,0 +1,836 @@
...
+You can refer to the following <vendor-name><product-name> documentation and
+Cloud VPN documentation for additional information about both products.
+
+### GCP documentation
+
+To learn more about GCP networking, see the following documents:
+
+-  [VPC Networks](https://cloud.google.com/vpc/docs)
+-  [Cloud VPN Overview](https://cloud.google.com/compute/docs/vpn/overview)
+-  [Creating Route-based VPNs](https://cloud.google.com/vpn/docs/how-to/creating-route-based-vpns)
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Since this is HA VPN, I'd remove the Classic VPN Route-based and Policy-based references and add the Cloud VPN topologies page: https://cloud.google.com/vpn/docs/concepts/topologies.

I'd also add the page to check VPN status, which includes checking for HA:
https://cloud.google.com/vpn/docs/how-to/checking-vpn-status

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 26]
```diff
@@ -0,0 +1,836 @@
...
+-  [Creating Policy-based VPNs](https://cloud.google.com/vpn/docs/how-to/creating-policy-based-vpns)
+-  [Advanced Cloud VPN Configurations](https://cloud.google.com/vpn/docs/concepts/advanced)
+-  [Terraform template for HA VPN](https://www.terraform.io/docs/providers/google/r/compute_ha_vpn_gateway.html)
+-  [Troubleshooting Cloud VPN](https://cloud.google.com/compute/docs/vpn/troubleshooting)
+
+### \<vendor-name>\<product-name> documentation
+
+For more product information on <vendor-name><product-name>, see the following
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Accessiblity issue: Consider using "refer to" or "read" rather than "see."
Thanks!

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 27]
```diff
@@ -0,0 +1,836 @@
+---
+title: Google Cloud HA VPN Interop Guide for [PRODUCT]
+description: Describes how to build site-to-site IPsec VPNs between HA VPN on Google Cloud Platform (GCP) and [VENDOR] [PRODUCT]
+author: [AUTHOR]
+tags: HA VPN, Cloud VPN, interop, [VENDOR], [PRODUCT]
+date_published: YYYY-mm-dd
+---
+
+**BEGIN: HOW TO USE THIS TEMPLATE**
```
[the remark this answers is not in the document]
Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: ack

[the remark this answers is not in the document]
Comment (jschneider11) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Thanks.

**Comment by ToddKopriva** [comment]
I'll review this in detial soon, but I didn't want to wait to give this crucial bit of information:

Our publishing system is veery aggressive about stripping things that might be security risks, and this includes stripping _everything_ between angle brackets.

So, for example, using <variable-name> as a way of indicating a variable won't work; the angle brackets and the text between them will not appear in the published document. That's why we use the [variable-name] construction for items published through the GCP Community system.

I need to make an update to the style guide to make this clear.

**Comment by jschneider11** [comment]
> I'll review this in detail soon, but I didn't want to wait to give this crucial bit of information:
> 
> Our publishing system is very aggressive about stripping things that might be security risks, and this includes stripping _everything_ between angle brackets.
> 
> So, for example, using <variable-name> as a way of indicating a variable won't work; the angle brackets and the text between them will not appear in the published document. That's why we use the [variable-name] construction for items published through the GCP Community system.
> 
> I need to make an update to the style guide to make this clear.

That fine with me then. I wasn't aware of that practice.

**Comment by ashishverm** [comment]
@ToddKopriva can you look at my last commit and confirm if this is way to represent variables ?
Should I make that change for the whole doc?

**Comment by ToddKopriva** [comment]
Yes, that is a good way to handle placeholders in this context. I agree with making such a change throughout the document.

**Comment by ashishverm** [comment]
> Yes, that is a good way to handle placeholders in this context. I agree with making such a change throughout the document.

**Comment by ashishverm** [comment]
> > Yes, that is a good way to handle placeholders in this context. I agree with making such a change throughout the document.

I checked the whole doc, all comments within <> should be removed from actual guides. 
As discussed offline I am leaving these as is and added a comment in "HOW TO USE THIS TEMPLATE" section.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 28]
```diff
@@ -0,0 +1,836 @@
...
+
+#### Create a custom VPC network
+
+If you haven't already, create a VPC network. These example instructions 
+create a [custom mode](https://cloud.google.com/vpc/docs/vpc#subnet-ranges) 
+VPC network with one subnet in one region and another subnet in another region.
+
+In the following commands, replace the options as noted below:
+
+[NETWORK] assign a network name.
```
[the remark this answers is not in the document]
Comment (ToddKopriva) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Documents published through the GCP Community system have very different limitations. Nothing that uses an angle bracket will pass through. Documents published through the GCP Communiyt site need to follow the swuare bracket pattern for placeholders that Ashish used here.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 29]
```diff
@@ -0,0 +1,836 @@
...
+[SUBNET_MODE] set as custom.
+[BGP_ROUTING_MODE] set as global.
+
+    gcloud compute networks create [NETWORK] \
+    --subnet-mode [SUBNET_MODE]  \
+    --bgp-routing-mode [BGP_ROUTING_MODE]
+
+The command output should look similar to the following example:
+
+    gcloud compute networks create network-a \
```
[the remark this answers is not in the document]
Comment (ToddKopriva) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: \<code\\> tags don't work in this system. Markdown only. Code should be indicated with backticks.

**Comment by ToddKopriva** [comment]
@jschneider11 , I'll hold off on merging this until you have given your approval.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 30]
```diff
@@ -0,0 +1,640 @@
...
+    --subnet-mode [SUBNET_MODE]  \
+    --bgp-routing-mode [BGP_ROUTING_MODE]
+
+Replace the placeholders as follows:
+
+- `[NETWORK]`: Assign a network name.
+- `[SUBNET_MODE]`: Set as `custom`.
+- `[BGP_ROUTING_MODE]`: Set as `global`.
+
+The command should look similar to the following example:
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: This is still incorrect. You can add the command output from the real guide as an example, or remove the output.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 31]
```diff
@@ -0,0 +1,640 @@
...
+    --network [NETWORK] \
+    --region [REGION_1] \
+    --range [RANGE_1]
+    
+    gcloud compute networks subnets create [SUBNET_NAME_2] \
+    --network [NETWORK] \
+    --region [REGION_2] \
+    --range [RANGE_2]
+
+The commands should look similar to the following example:
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Incorrect. Update with real output or remove.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 32]
```diff
@@ -0,0 +1,640 @@
...
+
+Configure firewall rules to allow inbound traffic from the on-premises
+network subnets:
+
+    gcloud compute firewall-rules create [VPN_RULE_NAME] \
+    --network [NETWORK] \
+    --allow tcp,udp,icmp \
+    --source-ranges [IP_ON_PREM_SUBNET]
+    
+The command should look similar to the following example:
```
Comment (jschneider11) [inline resources/templates/ha-vpn-interop-guide-template/index.md]: Incorrect. Update with real output or remove section.

**Comment by jschneider11** [comment]
If you need my approval, let's resolve this before I go on vacation next week.

File: resources/templates/ha-vpn-interop-guide-template/index.md (hunk 30 above)
Comment (ToddKopriva) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: I'll leave it to Ashish to decide what to do with these sections, but I am confused by your response, Julie.

The section begins by giving the command with some placeholders, then explains the placeholders, and then shows how the command would look with some example values instead of the palceholders. It's a little overly hand-holding for my tastes, but I don't understand why you are referring to it as incorrect or why you are referring to this as output.

Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Since this is a guide idea here is to give the actual command so users can just copy paste it on their terminal, not the output of the command as shown in official docs. I agree with your comment Todd, it is little too much for me as well but I am just trying to keep it similar to Classic VPN template.

Comment (jschneider11) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: I see, Todd. Ashish, I'll leave it up to whether or not you want to leave these examples in the guide.

Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Thanks both for the review.

File: resources/templates/ha-vpn-interop-guide-template/index.md [hunk 33]
```diff
@@ -0,0 +1,836 @@
...
+
+### Initial tasks
+
+Complete the following procedures before configuring a GCP HA VPN gateway and tunnel.
+
+#### Create a custom VPC network
+
+If you haven't already, create a VPC network. These example instructions 
+create a [custom mode](https://cloud.google.com/vpc/docs/vpc#subnet-ranges) 
+VPC network with one subnet in one region and another subnet in another region.
```
[the remark this answers is not in the document]
Comment (ashishverm) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: we have that added under "Configuration parameters and values" section. Does that suffice ?

[the remark this answers is not in the document]
Comment (jschneider11) [reply resources/templates/ha-vpn-interop-guide-template/index.md]: Yes, LGTM.
